crates/noirc_frontend/src/parser/parser.rs

```diff
@@ -185,14 +189,16 @@ fn function_definition(allow_self: bool) -> impl NoirParser<NoirFunction> {
         )
 }
 
-/// function_modifiers: 'unconstrained' 'open' | 'unconstrained' | 'open' | %empty
-///
-/// returns (is_unconstrained, is_open) for whether each keyword was present
-fn function_modifiers() -> impl NoirParser<(bool, bool)> {
+/// function_modifiers:  'open internal' | 'internal' | 'unconstrained' | 'open' | %empty
```

Comment:
Can we change it per my last comment? As-is this is not the correct grammar since 'open internal' is not a single token, and each of these tokens are OR'd together rather than parsed optionally and sequentially.

Reply:
Updated the values to show what noir would compile with. 

For example
```
// Fails with `Expected an end of input but found contract`. 
internal open fn quadruple(x: Field) -> pub Field { x * 4 }

// compiles
open internal fn quadruple(x: Field) -> pub Field { x * 4 }
```

Reply:
The grammar in the comment now is not correct with respect to what the function actually parses